OasisDEX/oasis-borrow
# Ades/assets landing pages feature toggle

# [Feature toggle for asset landing pages](https://app.shortcut.com/oazo-apps/story/3295/feature-toggle)
  
## Changes 👷‍♀️

- adds a new feature toggle functionality and exposes as a react hook
- adds a feature toggle `AssetLandingPages` for the new landing pages
- implements this feature in the index page of the app, showing a different UI when the feature is enabled
  
## How to test 🧪

### Test local enable/disable of feature toggle:

1. run the app
    - observe that the original homepage is showing
    - observe that the local storage contains a new key called `features` that contains a JSON
2. inside `features` JSON in local storage, change the value of `AssetLandingPages` from `false` to `true`
3. reload the homepage
    - observe that the string "AssetLandingPages feature enabled ✅" is displayed on the homepage

### Simulate a release

0. delete `features` from local storage
1. run the app
    - observe that the original homepage is showing
    - observe that in `features` in local storage, `AssetLandingPages` is set to `false`
2. inside `helpers/useFeatureToggle` set `configuredFeatures.AssetLandingPages` to `true`
3. recompile app and reload home page
    - observe that in `features` in local storage, `AssetLandingPages` remains set to `false`
    - observe that the string "AssetLandingPages feature enabled ✅" is displayed on the homepage

---

![image](https://user-images.githubusercontent.com/3783198/148851670-b451aa45-3df5-42f9-82e9-e611e193613e.png)
